File: src/julia/matmul.jl
function matgen(n)
	a = Array{Float64}(undef, n, n)
	tmp = 1.0 / n / n
	@inbounds for j = 1:n, i = 1:n
		a[i,j] = tmp * (i - j) * (i + j - 2)
	end
	return a
end

function matmul(n, a, b)
	c = zeros(Float64, n, n)
	for j = 1:n
		for k = 1:n
			for i = 1:n
				@inbounds begin
					c[i,j] += a[i,k] * b[k,j]
				end
			end
		end
	end
	return c
end

n = 1500
a = matgen(n)
b = matgen(n)
c = matmul(n, a, b)
println(c[(n>>1)+1,(n>>1)+1])

File: src/swift/matmul.swift
struct Matrix {
	var storage: [Float64]
	let cols_no: Int

	init(n: Int) {
		storage = Array(repeating: 0, count: n*n)
		cols_no = n
	}

	subscript(_ index: (Int, Int)) -> Float64 {
		get {
			let (i, j) = index
			return self.storage[i * self.cols_no + j]
		}
		set(newValue) {
			let (i, j) = index
			self.storage[i * self.cols_no + j] = newValue
		}
	}
}

func matgen(n: Int) -> Matrix {
	var a = Matrix(n: n)
	let tmp = 1.0 / Float64(n) / Float64(n);
	for i in 0...n-1 {
		for j in 0...n-1 {
			a[(i, j)] = tmp * Float64(i - j) * Float64(i + j)
		}
	}
	return a
}

func matmul(n: Int, a: Matrix, b: Matrix) -> Matrix {
	var c = Matrix(n: n)
	for i in 0...n-1 {
		for k in 0...n-1 {
			for j in 0...n-1 {
				c[(i, j)] += a[(i, k)] * b[(k, j)]
			}
		}
	}
	return c
}

var n = 1500
let a = matgen(n: n)
let b = matgen(n: n)
let c = matmul(n: n, a: a, b: b)
print(c[(n>>1, n>>1)])
